lms/djangoapps/branding/__init__.py

```diff
@@ -17,14 +17,26 @@
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
 
-def get_visible_courses():
+def get_visible_courses(org=None):
     """
-    Return the set of CourseDescriptors that should be visible in this branded instance
+    Return the set of CourseOverviews that should be visible in this branded instance
     """
-    filtered_by_org = microsite.get_value('course_org_filter')
-    courses = CourseOverview.get_all_courses(org=filtered_by_org)
+    microsite_org = microsite.get_value('course_org_filter')
+
+    if org and microsite_org:
+        # When called in the context of a microsite, return an empty result if the org
+        # passed by the caller does not match the designated microsite org.
+        courses = CourseOverview.get_all_courses(org=org) if org == microsite_org else []
+    else:
+        target_org = org or microsite_org
+        courses = CourseOverview.get_all_courses(org=target_org)
```

Comment:
Okay. @chrisndodge, I've cleaned up the logic used to decide which courses should be visible in a microsite. The end result is the same; all existing tests pass. Even so, would you mind taking a quick look at the changes in this module?